sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CodeGenerationSuite.scala

```diff
@@ -568,7 +568,6 @@ class CodeGenerationSuite extends SparkFunSuite with ExpressionEvalHelper {
     assert(refTerm.contains("scala.math.LowPriorityOrderingImplicits$$anon$"))
   }
 
-  // TODO (SPARK-35579): Fix this bug in janino and upgrade janino in Spark.
```

Comment:
Is there a test that should pass now after the fix?

Reply:
@srowen I think, when this bug was discovered we were at janino v3.1.4, ref [cloudfan's branch](https://github.com/cloud-fan/spark/tree/janino) for adding the UT in ignore mode, since the problem was only in 3.1.x branch of janino, we reverted back janino to 3.0.16 (3.0.x seems be deprecated as well [ref](https://github.com/janino-compiler/janino/pull/148#issuecomment-1085006136)). The [PR](https://github.com/apache/spark/pull/33302) to revert janino upgrade (3.1.4 -> 3.0.16), also enabled the UT since 3.0.x didn't had the issue.

Now when we bump back 3.1.7 i.e use janino 3.1.x and this test passes implies 3.1.7 has this bug fix, and kind of validates version bump is safe. WDYT ?

Reply:
Oh I see, sounds good